### PR TITLE
Adding table summarizing options needed for each rule

### PR DESCRIPTION
**Pull request by megancarney**

### CODE REVIEW

File: docs/source/ruletypes.rst [hunk 1]
```diff
@@ -5,6 +5,80 @@ Examples of several types of rule configuration can be found in the example_rule
...
++--------------------------------------------------------------------------------------------------------------------+
+|              FOR ALL RULES                                                                                         |
++==========================================+===========+=================================================+===========+
+| ``es_host`` (string)                     |           | ``buffer_time`` (time)                          | Optional  |
++------------------------------------------+           +-------------------------------------------------+           +
+| ``es_port`` (number)                     | Required  | ``query_delay`` (time)                          |           |
++------------------------------------------+-----------+-------------------------------------------------+           +
+| ``use_ssl`` (boolean, no default)        | Optional  | ``max_query_size`` (int, default 100k)          |           |
++------------------------------------------+           +-------------------------------------------------+-----------+
+| ``es_username`` (string, no default)     |           | ``filter`` (DSL filter, no default)             | Required  |
```
Comment (Qmando) [inline docs/source/ruletypes.rst]: Filter isn't actually required. (It was at one point, now it defaults to an empty list)

File: docs/source/ruletypes.rst [hunk 2]
```diff
@@ -5,6 +5,80 @@ Examples of several types of rule configuration can be found in the example_rule
...
++============================================+=====+===========+===========+========+===========+=======+==========+
+| ``compare_key`` (string, no default)       |     |    Req    |  Req      |    Req |           |       |          |
++--------------------------------------------+-----+-----------+-----------+--------+-----------+-------+----------+
+|``blacklist`` (list of strs, no default)    |     |   Req     |           |        |           |       |          |
++--------------------------------------------+-----+-----------+-----------+--------+-----------+-------+----------+
+|``whitelist`` (list of strs, no default)    |     |           |   Req     |        |           |       |          |
++--------------------------------------------+-----+-----------+-----------+--------+-----------+-------+----------+
+| ``ignore_null`` (boolean, no default)      |     |           |   Req     |  Req   |           |       |          |
++--------------------------------------------+-----+-----------+-----------+--------+-----------+-------+----------+
+| ``query_key`` (string, no default)         |     |           |           |   Req  |           |  Opt  |          |
```
Comment (Qmando) [inline docs/source/ruletypes.rst]: Also optional for Frequency

File: docs/source/ruletypes.rst [hunk 3]
```diff
@@ -5,6 +5,80 @@ Examples of several types of rule configuration can be found in the example_rule
...
++--------------------------------------------+-----+-----------+-----------+--------+-----------+-------+----------+
+|``use_count_query`` (boolean, no default)   |     |           |           |        |     Opt   | Opt   | Opt      |
+|                                            |     |           |           |        |           |       |          | 
+|``doc_type`` (string, no default)           |     |           |           |        |           |       |          | 
++--------------------------------------------+-----+-----------+-----------+--------+-----------+-------+----------+
+|``use_terms_query`` (boolean, no default)   |     |           |           |        |     Opt   | Opt   |          |
+|                                            |     |           |           |        |           |       |          | 
+|``doc_type`` (string, no default)           |     |           |           |        |           |       |          | 
+|                                            |     |           |           |        |           |       |          | 
+|``query_key`` (string, no default)          |     |           |           |        |           |       |          |
```
Comment (Qmando) [inline docs/source/ruletypes.rst]: Another to go under use_terms_query
``terms_size`` (Int, default 50)

This was just added to the docs in https://github.com/Yelp/elastalert/pull/84/files

File: docs/source/ruletypes.rst (hunk 2 above)
Comment (megancarney) [inline docs/source/ruletypes.rst]: What configuration option is optional? timeframe? num_events?

File: docs/source/ruletypes.rst (hunk 3 above)
Comment (megancarney) [inline docs/source/ruletypes.rst]: Done.

**Pull request merged**